sdk/eventhub/azure-eventhubs/azure/eventhub/_common.py

```diff
@@ -55,27 +55,44 @@ def __init__(self, body=None):
         self.message.annotations = {}
         self.message.application_properties = {}
 
+    def __repr__(self):
+        # type: () -> str
+        try:
+            body_str = self.body_as_str()
+        except:  # pylint: disable=bare-except
+            body_str = "<read-error>"
+        event_repr = "body='{}'".format(body_str)
+        try:
+            event_repr += ", properties={}".format(self.properties)
+            event_repr += ", offset={}".format(self.offset)
+            event_repr += ", sequence_number={}".format(self.sequence_number)
+            event_repr += ", partition_key={!r}".format(self.partition_key)
+            event_repr += ", enqueued_time={!r}".format(self.enqueued_time)
+        except:  # pylint: disable=bare-except
```

Comment:
Depending on which you expect it to be unhappy about, would it not be more robust (if a bit more verbose) to wrap those individually so the output still contains whatever data comes after? (Especially given \_\_repr\_\_ unambiguity aspirations; although I would roughly extend this comment to \_\_str\_\_ below as well)

This is mostly moot if it's really only enqueued_time, however, outside of a broad handwaving around precise exception capturing precision. 

Now, let my self-guilt for leaving this sort of a comment on a \_\_repr\_\_ commence :P

Reply:
I think this would probably be overkill for str, given that I really never expect these to fail. However I have updated repr accordingly due to the mentioned "unambiguity aspirations". :)